src/Timeline.js

```diff
@@ -42,8 +47,11 @@ export default class Timeline extends Axis {
     };
     this._shape = "Rect";
     this._shapeConfig = Object.assign({}, this._shapeConfig, {
-      height: d => d.tick ? 10 : 0,
-      width: d => d.tick ? this._domain.map(t => date(t).getTime()).includes(d.id) ? 2 : 1 : 0
+      labelBounds: d => this._buttonBehavior === "buttons" ? {x: -20, y: -5, width: 40, height: this._buttonHeight} : d.labelBounds,
+      fill: d => this._buttonBehavior === "buttons" ? "#EEE" : "#444",
+      height: d => this._buttonBehavior === "buttons" ? this._buttonHeight : d.tick ? 10 : 0,
+      width: d => this._buttonBehavior === "buttons" ? this._width / this._availableTicks.length : d.tick ? this._domain.map(t => date(t).getTime()).includes(d.id) ? 2 : 1 : 0,
```

Comment:
In the past, I used `this._buttonBehaviorCurrent` without success.

Reply:
the reason `this._buttonBehaviorCurrent` is not working is because you are setting it AFTER the timeline renders (`super.render(callback);`). simply move it above that line and it should work 😄